shared/chat/conversation/input-area/normal/wallets-icon/index.tsx

```diff
@@ -4,40 +4,54 @@ import * as Styles from '../../../../../styles'
 
 export type WalletsIconProps = {
   isNew: boolean
+  loadWalletsData: () => void
   onSend: () => void
   onRequest: () => void
   size: number
   style?: Styles.StylesCrossPlatform
 }
-const WalletsIcon = (props: WalletsIconProps & Kb.OverlayParentProps) => (
-  <Kb.Box2
-    ref={props.setAttachmentRef}
-    direction="horizontal"
-    style={Styles.collapseStyles([styles.container, props.style])}
-  >
-    <Kb.Icon type="iconfont-dollar-sign" fontSize={props.size} onClick={props.toggleShowingMenu} />
-    {props.isNew && <Kb.Box style={styles.newBadge} />}
-    <Kb.FloatingMenu
-      closeOnSelect={true}
-      attachTo={props.getAttachmentRef}
-      items={[
-        {
-          newTag: props.isNew,
-          onClick: props.onSend,
-          title: 'Send Lumens (XLM)',
-        },
-        {
-          newTag: props.isNew,
-          onClick: props.onRequest,
-          title: 'Request Lumens (XLM)',
-        },
-      ]}
-      onHidden={props.toggleShowingMenu}
-      position="top right"
-      visible={props.showingMenu}
-    />
-  </Kb.Box2>
-)
+class WalletsIcon extends React.Component<WalletsIconProps & Kb.OverlayParentProps, {}> {
+  componentDidMount() {
+    // Load the wallet data at mount so we can determine the account ID of the default wallet
+    this.props.loadWalletsData()
```

Comment:
I think it makes more sense to call this RPC on wallet icon click and use a loading indicator + waiting key to handle the race condition issue. It feels a little weird to me to be prefetching data like this on a mount if we don't have to.

Reply:
Where would you display the loading indicator for that operation? I guess instead of the contents of the popup? Just trying to imagine what UI we're going for here. 

![image](https://user-images.githubusercontent.com/5304541/62957878-9f46cb00-bdc3-11e9-9110-6243f912111f.png)